zion/src/test/scala/edu/uci/ics/cloudberry/zion/common/ConfigTest.scala

```diff
@@ -11,6 +11,17 @@ class ConfigTest extends Specification {
       Config.parseTimePair("43 minute") must_==(43 minutes)
       Config.parseTimePair("84 minutes") must_==(84 minutes)
     }
-
+    "parseFrameLengthLimit" in {
+      Config.parseFrameLengthLimit("20 B") must_==(8 * MemorySize.MB)
+      Config.parseFrameLengthLimit("0 KB") must_==(8 * MemorySize.MB)
+      Config.parseFrameLengthLimit("0.9 KB") must_==(8 * MemorySize.MB)
+      Config.parseFrameLengthLimit("1 KB") must_==(1 * MemorySize.KB)
+      Config.parseFrameLengthLimit("16 MB") must_==(16 * MemorySize.MB)
+      Config.parseFrameLengthLimit("16MB") must_==(16 * MemorySize.MB)
+      Config.parseFrameLengthLimit("1.5 GB") must_==(1.5 * MemorySize.GB)
+      Config.parseFrameLengthLimit("2 GB") must_==(Int.MaxValue)
+      Config.parseFrameLengthLimit("3 GB") must_==(8 * MemorySize.MB)
+      Config.parseFrameLengthLimit("5 TB") must_==(8 * MemorySize.MB)
```

Comment:
one principle about test case is that it should be meaningful and trackable when it fails. 

so in this case, you can structure them like following
```
"parseFrameLengthLimit should give 1KB if the given value is <= 1KB" in {
 ...
}
"parseFrameLengthLimit should parse the MB and GB if the given value is valid" in {
 ...
}
"parseFrameLengthLimit should use 2G as maxmumal value" in {
 ... 
}
```